### DISCUSSION TITLE
How does the default &pos filter work?

### DISCUSSION THREAD

**Discussion by Jens5395** [emporia-vue-local/esphome#10] [Q&A]
The &pos filter is "lambda: 'return max(x, 0.0f);'" .... I have been trying to find detailed info on the max() function. I can guess on the interpretation of max(x, 0.0f) but I have been unable to find a full explanation. The max filter documentation in ESPHome does not seem to agree with what is in &pos (specifically the 0.0f makes no sense and I would like to read up on the doc's)

In a related question - if I install a current sensor the wrong way around, can I not just the &invert filter? There have been some comments that just inverting the sensor data might produce incorrect data for some reason. I would like to clarify this since it is easier to use &invert than to take the electrical panel apart.

**Comment by flaviut** [comment]
`lambda: 'return max(x, 0.0f);'` is [the lambda filter](https://esphome.io/components/sensor/index.html#lambda). That means you've got to [look at the C++ documentation](https://en.cppreference.com/w/cpp/algorithm/max) for stuff in the body, since the body is c++ code. Max is `if (left > right) {return left} else {return right}`.

> can I not just the &invert filter?

Sure, go for it. I haven't tested it at all, so I can't personally recommend it. But you definitely won't damage anything.
